ludwig/models/llm.py

```diff
@@ -317,6 +329,7 @@ def train_loss(
         of_train_losses = {}
         for of_name, of_obj in self.output_features.items():
             _targets, _predictions = targets, predictions
+            ###SXK how can we extend this to other OutputFeature types
             if isinstance(of_obj, TextOutputFeature):
```

Comment:
It already supports Text and category output features, with more support coming shortly. This if block is specially for text features since they need to be handled separately since they have variable length output. Category/Numbers etc will have fixed length output so this provision doesn't need to be made.